#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material1 "wh"
      material2 "wh"
      coulombFriction [
        3
      ]
      bounce 1
    }
    ContactProperties {
      material2 "car"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.3237051267316949 0.8839920069845155 0.33730271643600135 1.6415432457712844
  position 2.3973238507067167 1.5791392133347195 -0.4084724229202199
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 3.2 3.2
}
Solid {
  translation 0.05 0.05 1.6
  children [
    DEF WALL_GEOMETRY Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        baseColorMap ImageTexture {
          url [
            "textures/red_brick_wall.jpg"
          ]
        }
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 3.2 0.1 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject USE WALL_GEOMETRY
  radarCrossSection 1
}
Solid {
  translation -0.05 0.05 -1.6
  children [
    USE WALL_GEOMETRY
  ]
  name "solid(3)"
  boundingObject USE WALL_GEOMETRY
  radarCrossSection 1
}
Solid {
  translation 1.6 0.05 -0.05
  rotation 0 1 0 1.5708003061004252
  children [
    USE WALL_GEOMETRY
  ]
  name "solid(4)"
  boundingObject USE WALL_GEOMETRY
  radarCrossSection 1
}
Solid {
  translation -1.6 0.05 0.0500071
  rotation 0 1 0 1.5708003061004252
  children [
    USE WALL_GEOMETRY
  ]
  name "solid(5)"
  boundingObject USE WALL_GEOMETRY
  radarCrossSection 1
}
WoodenBox {
  translation 0.55527 0.05 -0.804733
  rotation 0 1 0 -0.785395307179586
  size 0.1 0.1 1
  mass 0.2
}
WoodenBox {
  translation 1.04005 0.05 -0.291061
  rotation 0 1 0 -0.785395307179586
  name "wooden box(3)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.857646 0.05 -0.90928
  rotation 0 1 0 -0.785395307179586
  name "wooden box(11)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.166762 0.05 -0.415725
  rotation 0 1 0 -0.785395307179586
  name "wooden box(14)"
  size 0.4 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.876833 0.05 1.11655
  rotation 0 1 0 -0.785395307179586
  name "wooden box(7)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.019433 0.05 1.24322
  rotation 0 1 0 -0.785395307179586
  name "wooden box(9)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.759329 0.05 -0.353887
  rotation 0 1 0 -0.785395307179586
  name "wooden box(10)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -1.11928 0.05 1.04778
  name "wooden box(2)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -1.2669 0.05 -0.157006
  name "wooden box(8)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.315486 0.05 0.660504
  name "wooden box(12)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.489215 0.05 0.526245
  rotation 0 1 0 0.785398
  name "wooden box(4)"
  size 0.1 0.1 1
  mass 0.2
}
WoodenBox {
  translation -0.196284 0.05 -0.916552
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.639647 0.05 0.665603
  rotation 0 1 0 -0.785395307179586
  name "wooden box(6)"
  size 0.5 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 1.30393 0.05 0.0366482
  name "wooden box(17)"
  size 0.5 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.440097 0.05 0.00246688
  name "wooden box(13)"
  size 0.1 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.848817 0.05 0.444869
  rotation 0 -1 0 -0.7854053071795866
  name "wooden box(15)"
  size 0.3 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation -0.0960698 0.05 0.369738
  rotation 0 1 0 1.309
  name "wooden box(16)"
  size 0.3 0.1 0.1
  mass 0.2
}
WoodenBox {
  translation 0.733327 0.05 0.415116
  name "wooden box(5)"
  size 0.1 0.1 0.1
  mass 0.2
}
Robot {
  translation 0.611777 0.03 -0.431542
  rotation 0 1 0 0.523602
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 0.015 0.065
        minStop -0.8
        maxStop 0.8
      }
      device [
        DEF fl_rot_motor RotationalMotor {
          name "fl_rot_motor"
          maxVelocity 1
          minPosition -0.8
          maxPosition 0.8
        }
      ]
      endPoint Solid {
        translation 0.05 0.015 0.065
        children [
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
              anchor 0 -0.015 0.0015
            }
            device [
              RotationalMotor {
                name "wheel1"
              }
            ]
            endPoint Solid {
              translation 0 -0.015 0.0015000000000000013
              rotation 1 -1.5700895750559182e-16 -1.5700895750559181e-15 1.5707999999999813
              children [
                DEF Wheel1 Shape {
                  appearance PBRAppearance {
                    baseColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.0125
                    radius 0.015
                  }
                }
              ]
              name "solid1(3)"
              contactMaterial "wh"
              boundingObject USE Wheel1
              physics DEF Wheel_PH Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 0
                ]
              }
            }
          }
          DEF fl_rot_wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.005
            }
          }
        ]
        name "solid1"
        boundingObject USE fl_rot_wheel
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.05 0.015 0.005
        minStop -0.8
        maxStop 0.8
      }
      device [
        DEF fr_rot_motor RotationalMotor {
          name "fr_rot_motor"
          maxVelocity 1
          minPosition -0.8
          maxPosition 0.8
        }
      ]
      endPoint Solid {
        translation 0.05 0.015 0.004999999999999999
        rotation 0 1 0 3.141589988777128
        children [
          HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
              anchor 0 -0.015 0.0015
            }
            device [
              RotationalMotor {
                name "wheel2"
              }
            ]
            endPoint Solid {
              translation 0 -0.015 0.0015000000000000013
              rotation 1 -1.491585096303113e-15 2.590647798842249e-15 1.5707999999999938
              children [
                DEF Wheel2 Shape {
                  appearance PBRAppearance {
                    baseColor 0 1 0
                  }
                  geometry Cylinder {
                    height 0.0125
                    radius 0.015
                  }
                }
              ]
              name "solid1(3)"
              contactMaterial "wh"
              boundingObject USE Wheel2
              physics DEF Wheel_PH Physics {
                density -1
                mass 1
                centerOfMass [
                  0 0 0
                ]
              }
            }
          }
          DEF fr_rot_wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 1
            }
            geometry Cylinder {
              height 0.01
              radius 0.005
            }
          }
        ]
        name "solid2"
        boundingObject USE fr_rot_wheel
        physics Physics {
        }
      }
    }
    DEF ds_left DistanceSensor {
      translation 0.019992 0.02 0.06
      rotation 0 1 0 -2.880585307179586
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF ds_left DistanceSensor {
      translation 0.08 0.04 0.06
      rotation 0 1 0 -2.618785307179586
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Sphere {
            radius 0.005
          }
        }
      ]
      name "ds_left(1)"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF ds_left DistanceSensor {
      translation 0.119984 0.04 0.0700795
      rotation 0 1 0 -2.356995307179586
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Sphere {
            radius 0.005
          }
        }
      ]
      name "ds_left(2)"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF ds_right DistanceSensor {
      translation 0.020008 0.02 0.01
      rotation 0 -1 0 -2.879005307179586
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF ds_right DistanceSensor {
      translation 0.080008 0.04 0.00995224
      rotation 0 -1 0 -2.6171953071795864
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Sphere {
            radius 0.005
          }
        }
      ]
      name "ds_right(1)"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF ds_right DistanceSensor {
      translation 0.12 0.04 -7.96403e-05
      rotation 0 1 0 2.0936
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Sphere {
            radius 0.005
          }
        }
      ]
      name "ds_right(2)"
      boundingObject USE sensor
      physics Physics {
      }
      type "laser"
      aperture 0.3
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Mesh {
        url [
          "C:/Users/mak7i/OneDrive/Документы/БАП/4/Омир/6lab/car.obj"
        ]
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.19 0 0.0035
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF Wheel4 Solid {
        translation 0.19000773137326246 -8.878303846764288e-06 0.003500024285126692
        rotation 0.9999999994731809 -2.2953090948233826e-05 -2.2951991909553576e-05 1.5707999998480835
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Cylinder {
              height 0.0125
              radius 0.015
            }
          }
        ]
        name "solid1(2)"
        contactMaterial "wh"
        boundingObject USE Wheel
        physics USE Wheel_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor 0.19 0 0.0665
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF Wheel3 Solid {
        translation 0.19000607429800012 -6.975188093951521e-06 0.06650001911899929
        rotation 0.9999999994731807 -2.295311867875965e-05 -2.2951964564241733e-05 1.570799999793685
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
            }
            geometry Cylinder {
              height 0.0125
              radius 0.015
            }
          }
        ]
        name "solid3"
        contactMaterial "wh"
        boundingObject USE Wheel
        physics USE Wheel_PH
      }
    }
  ]
  contactMaterial "car"
  boundingObject USE Body
  physics Physics {
    density -1
    mass 3
    centerOfMass [
      0.125 0.01 0.035
    ]
  }
  controller "car_controller"
}
